packages/core/src/clients/middleware/userAgent/middleware.ts

```diff
@@ -0,0 +1,32 @@
+import { HttpRequest, HttpResponse } from '../../types/http';
+import { Middleware } from '../../types/core';
+
+export interface UserAgentOptions {
+	userAgentHeader?: string;
+	userAgentValue?: string;
+}
+
+// TODO: incorporate new user agent design
+export const userAgentMiddleware: Middleware<
+	HttpRequest,
+	HttpResponse,
+	UserAgentOptions
+> =
+	({
+		userAgentHeader = 'x-amz-user-agent',
+		userAgentValue = '',
+	}: UserAgentOptions) =>
+	next => {
+		return async function userAgentMiddleware(request) {
+			if (userAgentValue.trim().length === 0) {
+				return await next(request);
```

Comment:
Nit: Pretending we did have the lint rule for this, we'd want to assign the result before returning to be explicit about our intent (similar to ln28)
```suggestion
				const result = await next(request);
				return result;
```